### PR TITLE
SUSE15 sp2 changes

### PR DESCRIPTION
**Pull request by Himanshu-xilinx**
1. Added SUSE cmake packaging support
2. Updated xrtdeps.sh script for SUSE dependencies
3. Updated pkg_hw_platform.sh script for SUSE.

### CODE REVIEW

**Review by chvamshi-xilinx — APPROVED** [review]
Looks good.

File: src/runtime_src/core/tools/xbmgmt2/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+if ((${LINUX_FLAVOR} STREQUAL SUSE))
+set(Boost_USE_STATIC_LIBS OFF)               # find dynamic libraries
+else()
+set(Boost_USE_STATIC_LIBS ON)               # Only find static libraries
+endif()
+else()
+set(Boost_USE_STATIC_LIBS ON)               # Only find static libraries
+endif()
+set(Boost_USE_MULTITHREADED ON)             # Multi-threaded libraries
+
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbmgmt2/CMakeLists.txt:12]: Boost configuration must no be controlled in CMakeLists.txt post 2020.2.  If you notice this has changed in master as compared to 2020.2, we link dynamically with Boost for everything now.  Please remove.

File: src/runtime_src/core/tools/xbutil2/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+if ((${LINUX_FLAVOR} STREQUAL SUSE))
+set(Boost_USE_STATIC_LIBS OFF)               # find dynamic libraries
+else()
+set(Boost_USE_STATIC_LIBS ON)               # Only find static libraries
+endif()
+else()
+set(Boost_USE_STATIC_LIBS ON)               # Only find static libraries
+endif()
+set(Boost_USE_MULTITHREADED ON)             # Multi-threaded libraries
+
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbutil2/CMakeLists.txt:12]: See comment on xbmgmt2

**Review by stsoe — CHANGES_REQUESTED** [review]
Just two comments regarding Boost.   Otherwise looks good.

**Review by rozumx — CHANGES_REQUESTED** [review]
There are a lot of important changes in the pull request and yet the description of what was done is empty.   **_At least_** one would expect something like:
1. Added SUSE cmake packaging support
2. Updated xrtdeps.sh script for SUSE dependencies
3.  Updated xbutil2 & xbmgmt2 to use boost dynamic libraries for SUSE builds
4. Updated "classic" packager to be SUSE aware.

In addition, there are some unexpected changes to ulite.c, which "seems" out-of-place.

File: src/runtime_src/core/tools/xbmgmt2/CMakeLists.txt (hunk 1 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/xbmgmt2/CMakeLists.txt]: done

File: src/runtime_src/core/tools/xbutil2/CMakeLists.txt (hunk 2 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/tools/xbutil2/CMakeLists.txt]: done

**Review by stsoe — APPROVED** [review]

**Comment by xilinxgitops** [comment]
retest this please

**Comment by xilinxgitops** [comment]
retest this please

**Comment by xilinxgitops** [comment]
retest this please

**Comment by Himanshu-xilinx** [comment]
> There are a lot of important changes in the pull request and yet the description of what was done is empty. **_At least_** one would expect something like:
> 
> 1. Added SUSE cmake packaging support
> 2. Updated xrtdeps.sh script for SUSE dependencies
> 3. Updated xbutil2 & xbmgmt2 to use boost dynamic libraries for SUSE builds
> 4. Updated "classic" packager to be SUSE aware.
> 
> In addition, there are some unexpected changes to ulite.c, which "seems" out-of-place.

added comments @rozumx review

**Review by rozumx — APPROVED** [review]
Approved

**Pull request merged**